### PR TITLE
DOCS-11486: Create tasks for three uses cases of metadata assistant

### PR DESCRIPTION
**Pull request by fermujica**

### CODE REVIEW

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 1]
```diff
@@ -0,0 +1,161 @@
+= Propagate Metadata to Other Flows
+
+Studio enables you to propagate your flow's metadata to your referenced flows within your application.
+
+Depending on the metadata configured in your referenced flow, you might need to configure the specific metadata to propagate.
+
+== Propagate Metadata to a Flow Without Custom Metadata
+
+When you add a Flow Reference component to reference another flow with no custom metadata configured, Studio automatically propagates the metadata in your flow as custom metadata in your target flow.
```
Comment (agusmarracomulesoft) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: Should we put  `When you add a Flow Reference component and reference another flow` ....?

**Review by agusmarracomulesoft — COMMENTED** [review]
left only a wording comment, rest looks good to me

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 2]
```diff
@@ -0,0 +1,161 @@
+= Propagate Metadata to Other Flows
+
+Studio enables you to propagate your flow's metadata to your referenced flows within your application.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: Is this the same? Just trying to limit the use of "you(r)" in the sentence...lol.
```suggestion
Studio enables you to propagate your flow's metadata to other flows referenced within your application.
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+Depending on the metadata configured in your referenced flow, you might need to configure the specific metadata to propagate.
+
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc:4]: Yeah...in this sentence and the one before it, I wonder if the audience might get confused between "your flow" and "your referenced flow"? Maybe just look again at these sentences?

Comment (kbostrom) [reply modules/ROOT/pages/propagate-metadata-studio.adoc:4]: Looking below, I see you use "target": maybe "source flow" and "target flow"?
"Studio enables you to propagate metadata in your source flow to other, target flows referenced within your application. Depending on metadata already in the target flow, you might need to configure the specific metadata to propagate from the source flow."?

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+== Propagate Metadata to a Flow Without Custom Metadata
+
+When you add a Flow Reference component and reference another flow with no custom metadata configured, Studio automatically propagates the metadata in your flow as custom metadata in your target flow.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: Is this correct?
```suggestion
When you add a Flow Reference component to your source flow and reference a target flow with no custom metadata configured, Studio automatically propagates the metadata in your source flow as custom metadata in your target flow.
```

Comment (kbostrom) [reply modules/ROOT/pages/propagate-metadata-studio.adoc]: So...what is the actual task that follows? "To <do what?>:"

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+. In the taskbar at the top of the Anypoint Studio display, select *File* > *New* > *Mule Project*.
+. Type a name for this project, and select *Finish*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Type a name for your new project and select *Finish*.
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+. Drag two flow components to your empty canvas and name them `flow-1` and `flow-2` respectively.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag two flow components to your empty canvas and name them `flow-1` and `flow-2`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
++
+image::both-flows-metadata.png[]
+. In `flow-1`, drag a Set Payload component and configure the payload as:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: So, you are dragging the component to the flow, right? In that case:
```suggestion
. Drag a Set Payload component to `flow-1` and configure the payload:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
++
+image::set-payload-config.png[]
+. Drag a Logger component to `flow-2`, and configure it to log the `payload`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Logger component to `flow-2` and configure it to log `payload`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
++
+image::logger-payload-metadata.png[]
+. Drag a Flow Reference component to the end of `flow-1`, and configure it to reference _flow-2_:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Flow Reference component to the end of `flow-1` and configure it to reference _`flow-2`_:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
++
+image::flow-reference-metadata.png[]
+. Note that the metadata tab in `flow-2` is now populated with autogenerated metadata, and the input tab in the metadata explorer reflects the payload format that the Set Payload component determines in `flow-1`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: lowercase "explorer"?
```suggestion
. Note that the metadata tab in `flow-2` is now populated with autogenerated metadata, and the input tab in the metadata explorer reflects the payload format that the Set Payload component determines in `flow-1`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
++
+image::logger-metadata-case-1.png[]
+
+== Propagate Metadata When Extracting an Operation From Your flow to a New Flow.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
== Propagate Metadata When Extracting an Operation From Your Flow to a New Flow
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+When you extract an operation from your flow to a new flow or sub-flow, Studio enables you to configure the metadata that you want to propagate to the newly created flow:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
When you extract an operation from your source flow to propagate to a new target flow or subflow, Studio enables you to configure the metadata that you want to propagate to the newly created flow:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+
+. In the taskbar at the top of the Anypoint Studio display, select *File* > *New* > *Mule Project*.
+. Type a name for this project, and select *Finish*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Type a name for your new project and select *Finish*.
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+. Drag a flow component to your empty canvas and name it `flow-1`.
+. Drag a Set Payload component and configure the payload as:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Set Payload component to `flow-1` and configure the payload:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
++
+image::set-payload-flow-1.png[]
+. Drag a Logger component to the end of `flow-1`, and configure it to log `#[payload.value.departureDate]`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Logger component to the end of `flow-1` and configure it to log `#[payload.value.departureDate]`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
++
+image::logger-metadata-case-2.png[]
+. Right-click your Logger component and select *Extract To...* > *Flow*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Right-click your Logger component and select *Extract To...* > *Flow*:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
++
+image::extract-operation-metadata.png[]
+. Name your target flow `flow-2`.
+. Select *Metadata Preferences* to configure the input and output metadata. +
+Note that Studio automatically recognizes the metadata of your flow.
++
+image::extract-flow-metadata.png[]
+. Select *OK*.
+. Your Logger component is now in `flow-2`, the metadata tab is populated with autogenerated metadata, and the input tab in the metadata explorer reflects the payload format that the Set Payload component configures in `flow-1`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Your Logger component is now in `flow-2`, the metadata tab is populated with autogenerated metadata, and the input tab in the metadata explorer reflects the payload format that the Set Payload component configures in `flow-1`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
++
+image::extract-flow-metadata-final.png[]
+
+
+== Propagate Metadata to a Flow With Custom Metadata Already Configured
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
== Propagate Metadata to a Flow with Custom Metadata Already Configured
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+
+When you add a Flow Reference component and reference another flow that has custom metadata already configured, Studio prompts you to select the metadata that you want to propagate.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
When you add a Flow Reference component to your source flow and reference another, target flow that has custom metadata already configured, Studio prompts you to select the metadata that you want to propagate:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+
+. In the taskbar at the top of the Anypoint Studio display, select *File* > *New* > *Mule Project*.
+. Type a name for this project, and select *Finish*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Type a name for your new project and select *Finish*.
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+. Drag two flow components to your empty canvas and name them `flow-1` and `flow-2` respectively.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag two flow components to your empty canvas and name them `flow-1` and `flow-2`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
++
+image::both-flows-metadata.png[]
+. In `flow-1`, drag a Set Payload component and configure the payload as:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Set Payload component to `flow-1` and configure the payload:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
++
+image::set-payload-config.png[]
+. Create a JSON file with the name _arrival-type-example.json_ in your `src/main/resources/examples` directory with the following format:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Create a JSON file with the name _`arrival-type-example.json`_ in your `src/main/resources/examples` directory with the following format:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+    "ID": 1,
+    "arrivalDate": "2016/12/20",
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
+. Select `flow-2` in your canvas, select the *Metadata* tab, and *Add Metadata*:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Select `flow-2` > *Metadata* tab > *Add Metadata*:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 25]
```diff
@@ -0,0 +1,161 @@
...
++
+image::add-metadata-flow-2.png[]
+. Select *Input: Payload*, and then select the *Edit*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Select *Input: Payload* > *Edit*:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 26]
```diff
@@ -0,0 +1,161 @@
...
++
+image::configure-input-metadata.png[]
+.  Select *Add*, create a new metadata type called _arrivalType_, and select *Create type*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: Is there a reason for the italics? Seems like a lot of emphasis.
```suggestion
.  Select *Add*, create a new metadata type called _`arrivalType`_, and select *Create type*:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 27]
```diff
@@ -0,0 +1,161 @@
...
++
+image::create-metadata-type.png[]
+. Select the type *JSON* > *Example* and select the *...* button to browse to `src/main/resources/examples/arrival-type-example.json`
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Select the type *JSON* > *Example* and select the *...* button to browse to `src/main/resources/examples/arrival-type-example.json`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 28]
```diff
@@ -0,0 +1,161 @@
...
++
+image::create-json-metadata-type.png[]
+. Click *Select*.
+. Drag a Logger component to `flow-2`, and configure it to log the `payload`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Logger component to `flow-2` and configure it to log `payload`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 29]
```diff
@@ -0,0 +1,161 @@
...
++
+image::logger-payload-metadata.png[]
+. Drag a Flow Reference component to the end of `flow-1`, and configure it to reference _flow-2_:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Drag a Flow Reference component to the end of `flow-1` and configure it to reference _`flow-2`_:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 30]
```diff
@@ -0,0 +1,161 @@
...
++
+image::flow-reference-metadata.png[]
+. Select the Flow Reference component *Metadata* tab, and select *Set Metadata*.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
. Select the Flow Reference component *Metadata* tab, and select *Set Metadata*:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 31]
```diff
@@ -0,0 +1,161 @@
...
++
+image::set-metadata-example.png[]
++
+Studio warns you that setting the metadata from `flow-1` will overwrite the existing `arrivalType` metadata type configured for `flow-2`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
Studio warns you that setting the metadata from `flow-1` will overwrite the existing `arrivalType` metadata type configured for `flow-2`:
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 32]
```diff
@@ -0,0 +1,161 @@
...
++
+image::overwrite-metadata.png[]
+. Select *Customize* to configure the desired metadata to propagate to the flow:
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]: Colon only if the end of its sentence leads directly to the image, without any text between.
```suggestion
. Select *Customize* to configure the desired metadata to propagate to the flow.
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc [hunk 33]
```diff
@@ -0,0 +1,161 @@
...
++
+Note that Studio recognizes the actual metadata as the one set in `flow-1`, and the expected metadata as the one configured in `flow-2`.
```
Comment (kbostrom) [inline modules/ROOT/pages/propagate-metadata-studio.adoc]:
```suggestion
Note that Studio recognizes the actual metadata as that set in `flow-1`, and the expected metadata as that configured in `flow-2`.
```

File: modules/ROOT/pages/propagate-metadata-studio.adoc (hunk 32 above)
Comment (kbostrom) [reply modules/ROOT/pages/propagate-metadata-studio.adoc]: (You can avoid this by moving the second sentence to follow the image. See also Lines 76-79.)

**Review by kbostrom — APPROVED** [review]
CE 3 new examples

**Pull request merged**